### ISSUE TITLE
Create ingestion mechanism for RECAP content

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#585]
We need to figure out how to get the latest content from the RECAP server. I've done this a few times now, but we need to automate it and wrap it up with a cron job. This is pretty important, but I don't think the new RECAP search needs to wait for it.

**Comment by mlissner** [comment]
This was a big one, and it is now done, more or less, bringing #594 closer to completion. The new scraper is pretty sophisticated. It:

 - Gets a list of changed or modified items from the recap server (I tweaked the existing endpoint to make this work, and disabled the existing archive.recapthelaw.org scraper).
 - Groups the changed items into dockets, then downloads the docket XML file and the PDFs for a docket.
 - Parses the XML file, creating or updating the items in the DB.
 - For any new PDF, it gets its page numbers counted and the content extracted.
 - Once all of the above is complete, it gets added to the search index.

Now, what makes this all fun is that all of the above happens asynchronously like so:

 - A group of download tasks are spun off for every item in a docket that needs downloading. Those process in parallel.
 - When those all complete, a new task is created to do the parsing.
 - When it completes, a task is spun off to do the document extraction and OCR for each item in the docket in a loop. Ideally, this should be done in parallel for each document, but instead it's done in parallel for each docket. I couldn't figure out how to have a group of extraction items as an item in a chain. These are run in low priority, since they take time.
 - When extraction is complete, a new task does the save to the search engine.

In Celery speak, this is:

    chord(download_tasks)(chain(parse, extract_all, save))

Anyway, I've got the first run going now, and subsequent runs will trigger every four hours. I'd like this to be more frequent, but parsing documents takes time even if only a single item has changed, so it's best to do them in bundles.
